### PR TITLE
Adding API Server BOSH job

### PR DESCRIPTION
**Pull request by subhankarc**
- Added lock resource group and editing deployment resource

- Added defaultbackup resource

- Removed the old boshbackuprestore resource

- Removed the customstatus field

- Defaulted the state label to state value always

### CODE REVIEW

**Comment by subhankarc** [comment]
Dear reviewers,
Sorry for the huge PR.
The source mainly contains vendor files which are actually golang project dependencies. In the review, you can ignore them. Only thing which needed to be reviewed are under `jobs`, `packages`, `defaultbackup_types.go`, `director_types.go`, `deploymentlock_types.go`

Other files are autogenerated by the tool `apiserver-boot`

File: packages/service-fabrik-apiserver/packaging [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+# Build API Server package
+echo "Building API Server..."
+PACKAGE_NAME=github.com/cloudfoundry-incubator/service-fabrik-apiserver
+mkdir -p ${BOSH_INSTALL_TARGET}/src/${PACKAGE_NAME}
+cp -a ${BOSH_COMPILE_TARGET}/${PACKAGE_NAME}/* ${BOSH_INSTALL_TARGET}/src/${PACKAGE_NAME}
+export GOPATH=${BOSH_INSTALL_TARGET}
+go build -o ${BOSH_INSTALL_TARGET}/src/${PACKAGE_NAME}/bin/apiserver ${BOSH_INSTALL_TARGET}/src/${PACKAGE_NAME}/cmd/apiserver/main.go
+go build -o ${BOSH_INSTALL_TARGET}/src/${PACKAGE_NAME}/bin/controller-manager ${BOSH_INSTALL_TARGET}/src/${PACKAGE_NAME}/cmd/controller-manager/main.go
+# go install ${PACKAGE_NAME}
```
Comment (kamath-prasad) [inline packages/service-fabrik-apiserver/packaging]: pls remove commented code.

Comment (subhankarc) [reply packages/service-fabrik-apiserver/packaging]: Done

**Comment by subhankarc** [comment]
@kamath-prasad Can we merge this PR, as this is a disjoint PR and has no effect on the existing code.

If we merge, it will help in testing, as currently there are two PRs for boshrelease

**Pull request merged**